### ISSUE TITLE
[BUG] Configuration error in Crove's update document action when document id from previous step is used

### ISSUE THREAD

**Issue by croveapp** [PipedreamHQ/pipedream#3302]
**Describe the bug**
While configuring update document action of Crove's integration if we use the document id from the previous step of the workflow it shows some configuration error. In update document action dynamic fields are being populated with the help of entered document id, but when instead of manually adding the document id we use document id from the previous step no dynamic fields are populated and a configuration error is shown.

**To Reproduce**
1. Set up a new workflow and create a trigger from crove app e.g. document created trigger of crove app.
2. After testing the trigger create a new update document action of crove app.
3. While configuring update document action use document id from the previous step that is from document created trigger.
4. Instead of loading the dynamic fields it will a show a configuration error.

**Expected behavior**
When using the document id from previous step of workflow instead of showing configuration error it should load the dynamic fields.

**Screenshots**
Screenshots from the workflow
![image](https://user-images.githubusercontent.com/106083306/176959705-59b26d16-0964-4a96-9ffe-113758a4564a.png)

**Desktop**
 - OS: Windows 11
 - Browser: Chrome
 - Version: 103.0.5060.66

**Comment by dannyroosevelt** [comment]
I actually can't get the prop fields after Document ID to load at all, even when selecting an existing document from the list, via async options. @croveapp are you able to get the `update_document` action working when _not_ referencing a dynamic ID?

**Comment by croveapp** [comment]
There needs to be some variables in your document for the fields to load dynamically. You can create variables in your document from crove document editor. You can refer this on how to create variables in documents [https://help.crove.app/variables/adding-variables/](url)

@dannyroosevelt

**Comment by dannyroosevelt** [comment]
I'm running into a strange condition where I've now added variables to my document but when selecting the document using that action, no other props are rendering for me and I'm still able to test the step:
![Screenshot by Dropbox Capture](https://user-images.githubusercontent.com/20780753/176966614-65ae8a69-de01-4d1b-b34a-1161f6991eba.png)

In this case I am expecting my variables to load, or probably to receive a `[ConfigurationError](https://pipedream.com/docs/code/nodejs/#configuration-error)` with a pointer on how to fix the configuration.

**Comment by croveapp** [comment]
Have you added variable to your document as well ? You have to drag the variables from the right side bar into your document and then save the document. 
Like this 
![image](https://user-images.githubusercontent.com/106083306/176967491-3f563e8c-d019-4a73-9d3b-b3d914c04a52.png)

The dynamic field will load in the workflow like this
![image](https://user-images.githubusercontent.com/106083306/176967595-13192eea-da62-46a9-99ed-40158c15a369.png)

@dannyroosevelt

**Comment by dannyroosevelt** [comment]
Got it, thanks : )

Regarding your original issue, this is a known constraint right now, but I'm curious to understand more about your use case. If you're referencing the `documentId` dynamically via the inbound trigger, wouldn't you expect the configuration to change across executions? Can you you tell me more about what you're trying to do?

**Comment by croveapp** [comment]
Yes, so in crove we can generate new documents from the templates the document generated have all the variables as defined in the template. We have a trigger document_created which is fired whenever a new document from the template is generated. We want to fire the update_document action whenever document created trigger is triggered. As the generated document have same variables as defined in template the variables will be same across all the executions.

@dannyroosevelt

**Comment by dannyroosevelt** [comment]
Got it, thanks. We're chatting internally about the best way to approach your use case. Stay tuned.

**Comment by MrZxzx09** [comment]
need link plaese hheee..

**Comment by croveapp** [comment]
@dannyroosevelt do we have any update?

**Comment by croveapp** [comment]
@dannyroosevelt  was there any progress

**Comment by dannyroosevelt** [comment]
Hi there, unfortunately I don't have an update for you on this right now. As soon as we do we'll post updates to this ticket.
